lib/internal/util.js

```diff
@@ -182,8 +182,7 @@ function deprecate(fn, msg, code, useEmitSync, modifyPrototype = true) {
   );
 
   function deprecated(...args) {
-    // TODO(joyeecheung): use getOptionValue('--no-deprecation') instead.
-    if (!process.noDeprecation) {
+    if (!getOptionValue('--no-deprecation')) {
```

Comment:
Unfortunatley I don't think we can change this in this way. The `process.noDeprecate` property can be set dynamically at runtime and there are applications out there that take advantage of it. This change would end up breaking those existing uses. The irony here would be that we'd likely need to deprecate the `process.noDeprecation` flag.

Reply:
Yeah, in this case I think the TODO comment is just incorrect. As long as the option and the `process.noDeprecation` flag are compatible with each other and non-contradictory, then I would just remove the TODO comment and leave this as is.

Reply:
I think the TODO comment was left during refactoring when I was trying to figure out whether it's deprecatable at all - but at this point I am inclined to think that it is not as this shows up quite often in GitHub code search and also I am fairly certain there was one point I had to tell others to use it when there's no other alternative to avoid triggering excessive runtime warnings.